src/api-latency.ts

```diff
@@ -0,0 +1,60 @@
+import { NextFunction, Request, Response } from 'express';
+import * as onFinished from 'on-finished';
+
+import { MetricsGatherer } from './metrics-gatherer';
+
+const reqEndState = (req: Request): string => {
+	if (req.aborted) {
+		return 'aborted';
+	}
+	if (req.queueFailState) {
+		return req.queueFailState;
+	}
```

Comment:
What is this `queueFailState`? I can't see anywhere that sets it

Reply:
@Page-  this provides for the behaviour we currently have in the API, and doesn't do anything if the property isn't present. Ought to have annotated that. This PR goes with the (WIP) branch: https://github.com/balena-io/balena-api/pull/2510/files

Reply:
Hold up, I'm also failing to find this in `balena-api`. Not sure if you also meant that originally, but... this appears to have been a hopeful properly coded-for, but not in fact anywhere as-yet *added* by any code, to the request object, in the case of a timeout. This has to do with the fact that it's difficult to recover the underlying `express.Request` object when we timeout in `promise-queue`:

https://github.com/balena-io-modules/promise-queue/blob/master/src/index.ts#L81

Reply:
Similarly for rejections:

https://github.com/balena-io-modules/promise-queue/blob/master/src/index.ts#L107

Reply:
Yeah I couldn't find it anywhere so I wasn't really sure what it's meant to be